packages/web/src/apollo/suspense.tsx

```diff
@@ -290,6 +291,12 @@ export const RedwoodApolloProvider: React.FunctionComponent<{
     globalThis?.__REDWOOD__APOLLO_STATE ?? {}
   )
 
+  let useQueryHookSwap = useQuery
```

Comment:
@dac09 Please see this change. I had to do it because the two useQuery and useSuspenseQuery hooks aren't compatible once using Apollo Client 3.8.1 which I need for subscriptions to work due the the allowed content headers and multipart changes types in this up-to-date client.

IN the SSR PR before the client was 3.7.17 which didn't have the useQueryHook so somehow there was no type error.

I tried many TS gymnastics to get the GraphQLHookProvider and there Type Overrrides DefaultQueryTpe to be able to accept either hook, but TS was beyond me. Suspense didn't have loading, polling, etc etc.

This was a way to maybe get both in and have the types be happy together.

I don't know If suspense query works, but the tests do pass locally.

Open to other ideas and ways.

Reply:
Wouldn't it be easier to extend the hook provider to take useQuery and useSuspenseQuery? 

I think for now you can ignore the suspense setup as I don't have the bandwidth to debug both SSR+Streaming + subscriptions right now! 

SSR stuff isn't prod ready, and will get refactored many times - so best to keep it simple for now :)

Reply:
1. The why is quite simply useQuery and useSuspenseQuery types are incompatible once you use Apollo Client 3.8.1+ which is needed for subscriptions because of they support for the multi-part request.

Not doing such a swap has failing builds.

2.  extend the hook provider to take useQuery and useSuspenseQuery? 

Perhaps, but then how do cells work? It seemed the point of using useSuspenseQuery in place of useQuery is that cells would work still. If there is no useQuery for the cell, does it use useSuspenseQuery?


...

It feels like we are working atop each other and counterproductively trying to get to both our goals.

Reply:
> Not doing such a swap has failing builds.

Gotcha! 

> It feels like we are working atop each other and counterproductively trying to get to both our goals.

Yes you're right, and sorry about this. Let me see if I can unblock you!

Reply:
I just checked to see if I don't need the SSE link for multi channel subscriptions to work, but I do.

I'll look to see how better to configure/optional for those not using subscriptions.

Reply:
Here's one option: https://github.com/redwoodjs/redwood/pull/9052